crypto/mem.c

```diff
@@ -283,19 +124,20 @@ void *CRYPTO_malloc(int num, const char *file, int line)
     if (num <= 0)
         return NULL;
 
-    if (allow_customize)
-        allow_customize = 0;
-    if (malloc_debug_func != NULL) {
-        if (allow_customize_debug)
-            allow_customize_debug = 0;
-        malloc_debug_func(NULL, num, file, line, 0);
+#ifdef CRYPTO_MDEBUG
+    allow_customize = 0;
```

Comment:
I don't understand why allow_customize should remain at 1 if CRYPTO_MDEBUG is not set -- wouldn't changing the memory functions after an allocation has been made cause runtime issues whether or not debugging capabilities have been built in?
(Also appears in realloc.)